### PR TITLE
[WIP] Create API for OG oriented tests.

### PR DESCRIPTION
**Pull request by RoySegall**
#98

Just a mock up of the basic things.

### CODE REVIEW

File: tests/src/Kernel/Entity/GetEntityGroupsTest.php [hunk 1]
```diff
@@ -53,32 +56,22 @@ class GetEntityGroupsTest extends KernelTestBase {
...
-    $this->installEntitySchema('user');
-    $this->installEntitySchema('entity_test');
-    $this->installSchema('system', 'sequences');
+    // todo: check why this won't work with static variable.
+    $this->schemas = [
+      'config' => ['og'],
+      'entity' => ['og_membership', 'user', 'entity_test'],
+      'modules' => ['system' => ['sequences']],
+    ];
+
```
Comment (chx) [inline tests/src/Kernel/Entity/GetEntityGroupsTest.php]: This is not shorter really but changes from IDE integrated, autocomplete-able, ubiquitous methods to a proprietary mystery meat array. Strongly not recommended. (If you don't have an IDE with autocomplete, I recommend fixing that instead.)

File: src/OgTestTrait.php [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+   * Creating a group bundle.
+   *
+   * @param $entity_type_id
+   *   The entity ID.
+   *
+   * @return $this
+   */
+  protected function createGroupBundle($entity_type_id) {
+    $this->groupBundle = Unicode::strtolower($this->randomMachineName());
+    Og::groupManager()->addGroup($entity_type_id, $this->groupBundle);
```
Comment (chx) [inline src/OgTestTrait.php]: If we keep setting a property on the object (not really recommended, return it instead) this needs to be better documented.

File: tests/src/Kernel/Entity/GetEntityGroupsTest.php (hunk 1 above)
Comment (pfrenssen) [inline tests/src/Kernel/Entity/GetEntityGroupsTest.php:70]: :+1: I have to remember the "proprietary mystery meat array" phrase, that conveys it really well :)

**Comment by amitaibu** [comment]
I think it saves us little,  and adds more to learn. Let's keep being explicit for now in the tests.

Is that Ok with you Roy?

**Comment by RoySegall** [comment]
I'm still not complete about this one and how we can reduce the code we write in the setUp method so ignore this one for now.

**Comment by amitaibu** [comment]
Let's close this one for now. Seems it isn't much needed. Thank you still for working on it!

**Pull request closed without merging**